src/costmap_to_polygons.cpp

```diff
@@ -37,12 +37,71 @@
  *********************************************************************/
 
 #include <costmap_converter/costmap_to_polygons.h>
+#include <costmap_converter/misc.h>
 #include <boost/thread.hpp>
 #include <boost/thread/mutex.hpp>
 #include <pluginlib/class_list_macros.h>
 
 PLUGINLIB_EXPORT_CLASS(costmap_converter::CostmapToPolygonsDBSMCCH, costmap_converter::BaseCostmapToPolygons)
 
+namespace
+{
+
+/**
+ * @brief Douglas-Peucker Algorithm for fitting lines into ordered set of points
+ * 
+ * Douglas-Peucker Algorithm, see https://en.wikipedia.org/wiki/Ramer%E2%80%93Douglas%E2%80%93Peucker_algorithm
+ * 
+ * @param pointList the initial set of points
+ * @param epsilon distance criteria for removing point if it is smaller to the line segment than this
+ * @param result the simplified polygon
+ */
+void douglasPeucker(const std::vector<geometry_msgs::Point32> &pointList, double epsilon,
+                    std::vector<geometry_msgs::Point32> &result)
+{
+  if (pointList.size() <= 2)
+  {
+    result = pointList;
+    return;
+  }
+
+  // Find the point with the maximum distance from line between start and end
+  double dmax = 0.0;
```

Comment:
how about std::max_element?

Reply:
How to avoid recomputing the distance to the line multiple times in that case?
The Compare in max_element returns true if the first one is less than the second one.
But the comparison here happens not on the values of the vector but on the return of a function.
I could think of creating a temporary vector containing the distances. Does not sound like a good idea to me.
Am I missing something?

Reply:
nope, you are right: this would indeed result in twice as much calls to computeSquaredDistanceToLineSegment